jina/peapods/zmq/__init__.py

```diff
@@ -61,16 +63,25 @@ def __init__(
         self.is_polling_paused = False
         self.opened_socks = []  # this must be here for `close()`
         self.ctx, self.in_sock, self.out_sock, self.ctrl_sock = self._init_sockets()
+        self.in_sock_type = self.in_sock.type
+        if self.out_sock is not None:
+            self.out_sock_type = self.out_sock.type
+        else:
+            self.out_sock_type = zmq.DEALER
+        self.ctrl_sock_type = self.ctrl_sock.type
         self._register_pollin()
+        self.opened_socks.extend([self.in_sock, self.ctrl_sock])
+        if self.out_sock is not None:
+            self.opened_socks.append(self.out_sock)
 
-        self.opened_socks.extend([self.in_sock, self.out_sock, self.ctrl_sock])
+        self.out_sockets = {}
```

Comment:
I wonder if all sockets insight the current `Zmqlet` should go into another `Sockets` class, which cares about opening/closing (aka context managing) sockets, while the `Zmqlet` cares about the business logic.

Reply:
I would vote for this